### PR TITLE
ReplicaCAD datasources

### PR DESCRIPTION
**Pull request by aclegg3**
## Motivation and Context

Adds two new datasources for the ReplicaCAD dataset:
1. Interactive ReplicaCAD
2. ReplicaCAD variations with baked lighting.

NOTE: The data are not finalized.

## How Has This Been Tested

Adds a quickstart Colab/python script and adds to `test_examples.py`
`python examples/tutorials/nb_python/ReplicaCAD_quickstart.py`

## Types of changes

- [x] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have completed my CLA (see **CONTRIBUTING**)
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Review by jturner65 — APPROVED** [review]
LGTM

File: examples/tutorials/nb_python/ReplicaCAD_quickstart.py [hunk 1]
```diff
@@ -0,0 +1,339 @@
...
+# @markdown (double click to show code).
+
+# !curl -L https://raw.githubusercontent.com/facebookresearch/habitat-sim/master/examples/colab_utils/colab_install.sh | NIGHTLY=true bash -s
+# Note: for now we need to pull this branch to get the data downloader
+# %cd habitat-sim
+# !git fetch --unshallow
+# !git config remote.origin.fetch "+refs/heads/*:refs/remotes/origin/*"
+# !git config --get remote.origin.fetch
+# !git fetch --all
+# !git checkout --track origin/replica-cad
```
Comment (aclegg3) [inline examples/tutorials/nb_python/ReplicaCAD_quickstart.py:43]: Note: to be removed before merge.

File: tests/test_examples.py [hunk 2]
```diff
@@ -82,6 +82,24 @@ def test_vhacd_example(args):
     run_main_subproc(args)
 
 
+@pytest.mark.skipif(
+    not osp.exists("data/replica_cad/"),
```
Comment (mathfac) [inline tests/test_examples.py:86]: Should it be `data/scene_datasets/replica_cad/` as we have it for mp3d/gibson?

File: habitat_sim/utils/datasets_download.py [hunk 3]
```diff
@@ -69,6 +69,18 @@ def initialize_test_data_sources(data_path):
             "link": data_path + "scene_datasets/coda",
             "version": "1.0",
         },
+        "replica_cad_dataset": {
+            "source": "https://dl.fbaipublicfiles.com/habitat/ReplicaCAD/ReplicaCAD_dataset_v1.0.zip",
+            "package_name": "ReplicaCAD_dataset_v1.0.zip",
+            "link": data_path + "replica_cad",
```
Comment (mathfac) [inline habitat_sim/utils/datasets_download.py:75]: Should it be `"link": data_path + "scene_datasets/replica_cad",` as we have it for mp3d/gibson?

**Pull request merged**